core/internal/cltest/cltest.go

```diff
@@ -246,6 +257,25 @@ func NewApplicationWithConfig(t testing.TB, tc *TestConfig, flags ...string) (*T
 	}
 }
 
+func NewApplicationWithConfigAndKeyOnSimulatedBlockchain(
+	t testing.TB, tc *TestConfig, backend *backends.SimulatedBackend,
+	flags ...string) (app *TestApplication, cleanup func()) {
+	chainId := int(backend.Blockchain().Config().ChainID.Int64())
+	tc.Config.Set("ETH_CHAIN_ID", chainId)
+	app, appCleanup := NewApplicationWithConfigAndKey(t, tc, flags...)
+	var client SimulatedBackendClient
+	if txm, ok := app.Store.TxManager.(*store.EthTxManager); ok {
+		client = SimulatedBackendClient{b: backend, t: t, chainId: chainId}
+		txm.Client = &client
+	} else {
+		log.Panic("SimulatedBackend only works on EthTxManager")
+	}
+	// Clean out the mock registrations, since we don't need those...
+	app.EthMock.Responses = app.EthMock.Responses[:0]
```

Comment:
urgh... I wonder if we could avoid mocking the eth client entirely if this is the case?

Reply:
I guess we could, but the point of a mock is it's cheap, and I don't see much potential for confusion by keeping this around. I think at least this should go in a future story.

Reply:
Agreed, would be nice to avoid the initial setup all together if possible. Just needless extra steps, could see it being very unintuitive when we change ethMock and the simulated backend tests start breaking.